src/Microsoft.Restier.Core/ApiConfiguration.cs

```diff
@@ -65,5 +68,46 @@ public IServiceProvider ServiceProvider
         {
             return this.serviceProvider.GetService<T>();
         }
+
+        internal TaskCompletionSource<IEdmModel> CompeteModelGeneration(out Task<IEdmModel> running)
+        {
+            var source = new TaskCompletionSource<IEdmModel>(TaskCreationOptions.AttachedToParent);
+            var exist = Interlocked.CompareExchange(ref modelTask, source.Task, null);
+            if (exist != null)
+            {
+                running = exist;
+                source.SetCanceled();
+                return null;
+            }
+
+            source.Task.ContinueWith(
```

Comment:
You mean set the out running parameter? It's not necessary I think, since source.Task continues when SetResult or SetException is called in GetModelAsync, that will trigger the continuation here.